File: day01/day1.py
# testInput = ")())())"
from input import input


def find_floors(parens):
    height = 0
    for paren in parens:
        height += 1 if paren == "(" else -1
    print(height)


def find_basement(parens):
    height = 0
    for index, paren in enumerate(parens):
        height += 1 if paren == "(" else -1
        if height < 0:
            print(index + 1)
            return


find_floors(input)
find_basement(input)

File: day01/day1.js
import parens from './input.js';

function calcHeight(parens) {
    let height = 0;
    for (let i = 0; i < parens.length; i++) {
        if (parens[i] === '(') {
            height += 1;
        } else {
            height -= 1;
        }
    }
    return height;
}

function enterBasement(parens) {
    let height = 0;
    for (let i = 0; i < parens.length; i++) {
        if (parens[i] === '(') {
            height += 1;
        } else {
            height -= 1;
        }
        if (height === -1) {
            return i + 1;
        }
    }
    return null;
}

console.log(calcHeight(parens));
console.log(enterBasement(parens));
